starboard/shared/starboard/media/iamf_util_test.cc

```diff
@@ -0,0 +1,163 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/shared/starboard/media/iamf_util.h"
+
+#include <string>
+
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace starboard {
+namespace shared {
+namespace starboard {
+namespace media {
+namespace {
+
+TEST(IamfUtilTest, IsValid) {
+  std::string codec_param = "iamf.000.000.Opus";
+  IamfMimeUtil util(codec_param);
+  EXPECT_TRUE(util.is_valid());
+
+  codec_param = "iamf.000.000.fLaC";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_TRUE(util.is_valid());
+
+  codec_param = "iamf.000.000.ipcm";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_TRUE(util.is_valid());
+
+  codec_param = "iamf.000.000.mp4a.40.2";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_TRUE(util.is_valid());
+
+  // Invalid params
+  codec_param = "iamf.000.000.vorbis";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.000.00.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.000.999.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.999.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iacb.000.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.000.000.Opus.40.2";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.000.000.mp4a.40.3";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.000.000.mp4a.40.20";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.000.000..Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf000.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.00.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.000.00.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.0aa.000.Opus.";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.000.0aa.Opus.";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  codec_param = "iamf.0O0.000.Opus.";
```

Comment:
Consider removing the trailing '.' due to the reason mentioned in the above comment.

Also nit: "0O0" => "0X0", the former one can sometimes be hard for humans to know that the second character is not a 0.

Reply:
Added a comment explaining, but I can make that change if it's better.